endpoint/platform/wp77xx/wp77xx.c

```diff
@@ -0,0 +1,236 @@
+/*
+ * Copyright (C) 2019-2020 BiiLabs Co., Ltd. and Contributors
+ * All Rights Reserved.
+ * This is free software; you can redistribute it and/or modify it under the
+ * terms of the MIT license. A copy of the license can be found in the file
+ * "LICENSE" at the root of this distribution.
+ */
+
+#include <errno.h>
+#include <fcntl.h>
+#include <stdbool.h>
+#include <stdint.h>
+#include <stdio.h>
+#include <stdlib.h>
+#include <string.h>
+#include <termios.h>
+#include <time.h>
+#include <unistd.h>
+#include "endpoint/hal/device.h"
+
+#include "legato.h"
+
+#include "interfaces.h"
+#include "le_log.h"
+#include "le_mdmDefs_interface.h"
+#include "le_secStore_common.h"
+#include "le_sim_common.h"
+
+#define MAXLINE 1024
+
+#define CBC_IV_SIZE 16
+#define IMSI_LEN 15
+#define READ_BUFFER_SIZE 32
+#define DEFAULT_PORT "/dev/ttyHS0"
+
+static inline void register_wp77xx(void);
```

Comment:
It is not worthy to have two inline functions. Instead, you can implement within function `wp77xx_release` and `wp77xx_init`.